### PR TITLE
Cheq Reqs

### PR DESCRIPTION
**Pull request by alexandrawang1**
# [Issue 29](https://github.com/MathSoc/mathsoc-website/issues/29)

Redevelop https://mathsoc.uwaterloo.ca/cheque-request-information/ for the new site.

Acceptance criteria:

The page is rebuilt, containing all information from the version on the old site.

cheqreqs

## Notes
- the refund form is not available, but i wanted to keep the link there as a placeholder so its not forgotten. if we decide we don't need it, it can easily be removed
- the collapsible styling for the FAQ section is from the volunteer page, but the `collapsible` pug component is specific to the volunteer page

### CODE REVIEW

File: public/assets/sass/pages/resources/cheque-requests.scss [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    content: '\02795'; /* Unicode character for "plus" sign (+) */
+    font-size: 13px;
+    color: white;
+    float: right;
+    margin-left: 5px;
+}
+
+.active:after {
+    content: "\2796"; /* Unicode character for "minus" sign (-) */
+}
```
Comment (some-molasses) [inline public/assets/sass/pages/resources/cheque-requests.scss:54]: Rather than copying and pasting the styling for the collapsible component, we could instead import it from the existing `_collapsible.scss` file :eyes: 

```suggestion
@import "../../mixins/collapsible";
```

File: server/data/resources/cheque-requests.json [hunk 2]
```diff
@@ -0,0 +1,117 @@
+{
+  "title": "Cheque Request Forms",
+  "formLinks": [
+    {
+      "title": "Club Cheque Request Form",
+      "link": "https://mathsoc.uwaterloo.ca/wp-content/uploads/Club-Cheque-Request-Form.pdf"
+    },
+    {
+      "title": "MathSoc Cheque Request Form",
+      "link": "https://mathsoc.uwaterloo.ca/wp-content/uploads/MathSoc-Cheque-Request-Form-1.pdf"
```
Comment (some-molasses) [inline server/data/resources/cheque-requests.json:10]: We'll need to deal with moving these files to the rebuilt site before we migrate to this new version.  Not only will we need to have them on our own server, but we'll also need to give admins the ability to upload updated versions of the cheque request forms.

Could you create an issue to build an interface for updating these forms?  We don't necessarily need it as part of this PR, but it will be important to figure out soon.

File: views/pages/resources/cheque-requests.pug [hunk 3]
```diff
@@ -0,0 +1,38 @@
+extends /views/templates/frontend-base.pug
+
+append styles
+    link(href="/assets/css/pages/resources/cheque-requests.css" rel="stylesheet")
+append scripts
+    script(src="/assets/js/collapsible.js" defer)
+
+block body
+    +banner-image("/assets/img/banners/mathsoc-wall.jpeg")
```
Comment (some-molasses) [inline views/pages/resources/cheque-requests.pug:9]: Thoughts on whether this page would look better without a banner image at the top?

File: views/pages/resources/cheque-requests.pug [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+append scripts
+    script(src="/assets/js/collapsible.js" defer)
+
+block body
+    +banner-image("/assets/img/banners/mathsoc-wall.jpeg")
+    div(class="content")
+        section
+            h1 #{data.title}
+            each form in data.formLinks
+                a(href=form.link target="_blank" class="form-link")
```
Comment (some-molasses) [inline views/pages/resources/cheque-requests.pug:14]: Thoughts on making these visually look like buttons instead of the text-based links from the wordpress site?

(quick MS paint sketch below: )

File: server/data/resources/cheque-requests.json [hunk 5]
```diff
@@ -0,0 +1,117 @@
+{
+  "title": "Cheque Request Forms",
+  "formLinks": [
+    {
+      "title": "Club Cheque Request Form",
```
Comment (some-molasses) [inline server/data/resources/cheque-requests.json:5]: Can we rephrase this to "Download club cheque request form"?  For accessibility reasons, it's better if for any link or button, we clearly state the action that clicking on it will cause.

File: server/data/resources/cheque-requests.json [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  "otherForms": {
+    "header": "Other Forms",
+    "footnote": "This form is used to request a refund of the MathSoc fee. It must be filled out in the first month of the term.",
+    "forms": [
+      {
+        "title": "Academic Feedback Form",
+        "link": "https://docs.google.com/forms/d/e/1FAIpQLScl3yNAcyJO0v3z2Pot_PLnVtnsMGHrCFZm4eM90YF6no7SlQ/viewform"
+      },
+      {
+        "title": "Capital Improvements Fund Form",
```
Comment (some-molasses) [inline server/data/resources/cheque-requests.json:96]: Same comment here about adding 'Download'

**Review by some-molasses — CHANGES_REQUESTED** [review]
Thanks for building out this page!

**Review by ad-world — CHANGES_REQUESTED** [review]
could you add the `cheque-requests.json` file to our data validation pipeline?

File: server/data/resources/cheque-requests.json (hunk 2 above)
Comment (alexandrawang1) [reply server/data/resources/cheque-requests.json:10]: Good catch - I didn't notice that about the urls. Here's the issue I made: https://github.com/MathSoc/mathsoc-website/issues/136

Feel free to edit it, of course

File: server/data/resources/cheque-requests.json (hunk 5 above)
Comment (alexandrawang1) [reply server/data/resources/cheque-requests.json:5]: Done!

File: views/pages/resources/cheque-requests.pug (hunk 3 above)
Comment (alexandrawang1) [reply views/pages/resources/cheque-requests.pug:9]: Agreed, done

File: public/assets/sass/pages/resources/cheque-requests.scss (hunk 1 above)
Comment (alexandrawang1) [reply public/assets/sass/pages/resources/cheque-requests.scss:54]: Good catch, done!

File: server/data/resources/cheque-requests.json (hunk 6 above)
Comment (alexandrawang1) [reply server/data/resources/cheque-requests.json:96]: I added it for the cheque request forms, but would "Download" be appropriate here since these are google/office forms that will not be downloaded? Only the pdfs need to be downloaded to be submitted, I think

**Comment by alexandrawang1** [comment]
thanks, json added to validation

File: server/data/resources/cheque-requests.json (hunk 2 above)
Comment (some-molasses) [reply server/data/resources/cheque-requests.json:10]: 🙏

File: server/data/resources/cheque-requests.json (hunk 6 above)
Comment (some-molasses) [reply server/data/resources/cheque-requests.json:96]: oops yeah you're 100% right, nvm.

**Review by some-molasses — APPROVED** [review]

File: views/pages/resources/cheque-requests.pug (hunk 4 above)
Comment (alexandrawang1) [reply views/pages/resources/cheque-requests.pug:11]: I feel they're a little difficult to read, and might get chunky if we add more forms in the future. But otherwise I don't have a strong opinion and I can add them if that's what we want for design

**Review by ad-world — APPROVED** [review]
